main.go

```diff
@@ -95,13 +136,41 @@ func (g Gaz) Run() {
 
 		// register /version to return the build version
 		g.Router.HandleFunc("/version", VersionHTML).Methods("GET")
-
-		port := viper.GetInt("http.port")
-		Sugar.Infof("Starting http server on port %d", port)
-		err := http.ListenAndServe(fmt.Sprintf(":%d", port), g.Router)
+		Sugar.Infof("Starting HTTP server on :%d", g.httpPort)
+		err := http.Serve(*g.httpListener, g.Router)
 		if err != nil {
-			Sugar.Errorf("Cannot start HTTP server on :%d, %+v", port, err)
+			Sugar.Errorf("Cannot updater HTTP server on :%d, %+v", g.httpPort, err)
 			panic(err)
 		}
 	}()
 }
+
+func (g Gaz) Register(d *ServiceDefinition) (string, error) {
+	if g.ServiceDiscovery == nil {
+		return "", errors.New("no service registry configured")
+	}
+	return g.ServiceDiscovery.Register(d, g.httpPort)
+}
+
+func (g Gaz) DeRegister(serviceId string) error {
+	return g.ServiceDiscovery.DeRegister(serviceId)
+}
+
+func (g Gaz) Resolve(serviceName string) ([]ServiceEndpoint, error) {
+	return g.ServiceDiscovery.Resolve(serviceName)
+}
+
+func (g Gaz) GrpcDial(target string, opts ...grpc.DialOption) (*grpc.ClientConn, error) {
+	clientKeepAlive := grpc.WithKeepaliveParams(keepalive.ClientParameters{
+		Time:                15 * time.Second,
+		PermitWithoutStream: true,
+	})
+	options := make([]grpc.DialOption, len(opts)+3)
+	options[0] = grpc.WithBalancerName(roundrobin.Name)
+	options[1] = grpc.WithBackoffMaxDelay(5 * time.Second)
```

Comment:
This should be overridable somehow

Reply:
yes, it the clients put these options in the config, it will override the defaults